common/djangoapps/student/views.py

```diff
@@ -2718,6 +2718,77 @@ def validate_new_email(user, new_email):
         raise ValueError(_('An account with this e-mail already exists.'))
 
 
+def validate_social_link(social_field, new_social_link):
+    """
+    Given a new social link for a user, ensure that the link takes one of the
+    following forms:
+
+    1) A valid url that comes from the correct social site
+    2) A valid username
+    3) A blank value
+    """
+    # Check if the link contains a valid url to the core domain
+    required_url_stub = settings.SOCIAL_PLATFORMS[social_field]['url_stub']
+    contains_social_url = required_url_stub in new_social_link \
+                          and '.' not in new_social_link.replace(required_url_stub, '')
```

Comment:
I'm not sure of what this new second case is for and I think it is overkill.

Here's how I envision this code getting simpler:
1. Use original simple code: `contains_social_url = required_url_stub in new_social_link`.
2. If contains_social_url is True, have a method (or additional line) to pull out the username by taking everything after the last `/`.  If the url ends in a '/', first drop it.
3. Now, you should have a username (either from the url or passed in), and you can check if it is valid.
Voila.